### PR TITLE
run the application with kafka on docker (using colima)

### PR DESCRIPTION
**Pull request by bsneha90TW**
This repo runs code with Kafka running on docker.

Docker would be running on colima.

This setup has 1 zookeeper and 3 brokers.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    container_name: zookeeper-1
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 3181
+      ZOOKEEPER_TICK_TIME: 2000
+      ZOOKEEPER_INIT_LIMIT: 5
+      ZOOKEEPER_SYNC_LIMIT: 2
+    ports:
+      - "23181:3181"
+  kafka-workshop-1:
+    image: confluentinc/cp-kafka
```
Comment (akhil-ghatiki) [inline docker-compose.yml]:
```suggestion
    image: confluentinc/cp-kafka:latest
```

shall we add latest tag to this ?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      KAFKA_BROKER_ID: 1
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper-1:3181
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka-workshop-1:29092,PLAINTEXT_HOST://localhost:9092
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 100
+
+  kafka-workshop-2:
+    image: confluentinc/cp-kafka
```
Comment (akhil-ghatiki) [inline docker-compose.yml]:
```suggestion
    image: confluentinc/cp-kafka:latest
```

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      KAFKA_BROKER_ID: 2
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper-1:3181
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka-workshop-2:29093,PLAINTEXT_HOST://localhost:9093
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 100
+
+  kafka-workshop-3:
+    image: confluentinc/cp-kafka
```
Comment (akhil-ghatiki) [inline docker-compose.yml]:
```suggestion
    image: confluentinc/cp-kafka:latest
```

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    environment:
+      - KAFKA_CLUSTERS_0_NAME=kafka-workshop-1
+      - KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS=kafka-workshop-1:29092
+      - KAFKA_CLUSTERS_0_ZOOKEEPER=localhost:23181
+      - KAFKA_CLUSTERS_1_NAME=kafka-workshop-2
+      - KAFKA_CLUSTERS_1_BOOTSTRAPSERVERS=kafka-workshop-2:29093
+      - KAFKA_CLUSTERS_1_ZOOKEEPER=localhost:23181
+      - KAFKA_CLUSTERS_2_NAME=kafka-workshop-3
+      - KAFKA_CLUSTERS_2_BOOTSTRAPSERVERS=kafka-workshop-3:29094
+      - KAFKA_CLUSTERS_2_ZOOKEEPER=localhost:23181
```
Comment (akhil-ghatiki) [inline docker-compose.yml]: lets add a new line at the end of this file

File: docs/01.setup_kafka_with_docker.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+``
+HOMEBREW_NO_AUTO_UPDATE=1 brew install colima
+``
+
+### 2. Install docker client
+
+``
+HOMEBREW_NO_AUTO_UPDATE=1 brew install docker
+``
+
```
Comment (akhil-ghatiki) [inline docs/01.setup_kafka_with_docker.md:19]: can we add a step to install docker-compose ?
```suggestion
brew install docker-compose
```

File: docs/02.console_producers_and_consumers.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+The configuration of existing topics can be updates. Below is one example
+
+- update the min.insync.replicas config of the topic
+```shell script
+docker exec kafka-workshop-1 kafka-configs --bootstrap-server localhost:29092 --alter --entity-type topics --entity-name my-first-topic --add-config min.insync.replicas=3
+```
+
+**********************
+### 1.  Create a topic first beore producing/consuming messages from Kafka.
```
Comment (akhil-ghatiki) [inline docs/02.console_producers_and_consumers.md:45]: can we run commands in this section without docker exec ? In my machine, it was not able to connect to the broker.

File: docs/01.setup_kafka_with_docker.md
Comment (akhil-ghatiki) [inline docs/01.setup_kafka_with_docker.md:1]: Shall we rename this file to `setup_kafka_with_colima.md` ?

**Review by akhil-ghatiki — CHANGES_REQUESTED** [review]
requesting few small changes.

File: docs/01.setup_kafka_with_docker.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+`http://localhost:8095/`
+
+### 5. Stop kafka brokers and UI
+Once you are done with workshop, stop Kafka server using below command.
+
+`docker-compose -f docker-compose.yml down`
+
+## Steps to stop colima (and switch back to docker desktop if you have it)
+
+### 1. Stop colima using `colima stop`.
```
Comment (akhil-ghatiki) [inline docs/01.setup_kafka_with_docker.md]:
```suggestion
1. Stop colima using `colima stop`.
```
Just the styling

File: docs/01.setup_kafka_with_docker.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+### 2. Check the docker context, to make sure it is default (for Docker desktop users)
```
Comment (akhil-ghatiki) [inline docs/01.setup_kafka_with_docker.md]:
```suggestion
 2. Check the docker context, to make sure it is default (for Docker desktop users)
```
styling fix.

File: docs/01.setup_kafka_with_docker.md
Comment (bsneha90TW) [reply docs/01.setup_kafka_with_docker.md:1]: It is actually running with docker, just the runtime is colima?

File: docs/02.console_producers_and_consumers.md (hunk 6 above)
Comment (bsneha90TW) [reply docs/02.console_producers_and_consumers.md:45]: can we check this together? We want to run the commands on the docker container created in previous step. Why it was not able to connect for you, we can check.

**Review by akhil-ghatiki — APPROVED** [review]

**Pull request merged**